### PR TITLE
Worldpay: Update stored credentials logic

### PR DESCRIPTION
**Pull request by DustinHaefele**
Remote tests:
99 tests, 429 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

Unit tests:
116 tests, 658 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

Spreedly ticket: ECS-3238

### CODE REVIEW

File: lib/active_merchant/billing/gateways/worldpay.rb [hunk 1]
```diff
@@ -684,30 +684,27 @@ def add_stored_credential_options(xml, options = {})
...
-          reason = case options[:stored_credential][:reason_type]
-                   when 'installment' then 'INSTALMENT'
-                   when 'recurring' then 'RECURRING'
-                   when 'unscheduled' then 'UNSCHEDULED'
-                   end
-
-          xml.storedCredentials 'usage' => 'USED', 'merchantInitiatedReason' => reason do
-            xml.schemeTransactionIdentifier options[:stored_credential][:network_transaction_id] if options[:stored_credential][:network_transaction_id]
-          end
+        reason = case options[:stored_credential][:reason_type]
```
Comment (aenand) [inline lib/active_merchant/billing/gateways/worldpay.rb:687]: The previous code did not include this if it was the `initial_transaction`, did we want to preserve that logic?

File: lib/active_merchant/billing/gateways/worldpay.rb [hunk 2]
```diff
@@ -1027,6 +1024,15 @@ def eligible_for_0_auth?(payment_method, options = {})
       def card_holder_name(payment_method, options)
         test? && options[:execute_threed] && !options[:three_ds_version]&.start_with?('2') ? '3D' : payment_method.name
       end
+
+      def generate_stored_credential_params(is_initial_transaction, reason = nil)
+        customer_or_merchant = reason == 'RECURRING' && is_initial_transaction ? 'customerInitiatedReason' : 'merchantInitiatedReason'
```
Comment (aenand) [inline lib/active_merchant/billing/gateways/worldpay.rb:1029]: Should we use `stored_credential[:initiator]` for customer_or_merchant?

File: test/remote/gateways/remote_worldpay_test.rb [hunk 3]
```diff
@@ -521,6 +521,40 @@ def test_successful_auth_and_capture_with_normalized_stored_credential
     assert_success capture
   end
 
+  def test_successful_auth_and_capture_with_normalized_recurring_stored_credential
+    stored_credential_params = {
```
Comment (aenand) [inline test/remote/gateways/remote_worldpay_test.rb:525]: Fun fact, there is a way to invoke standardized stored creds params in active_merchant by calling `stored_credential`. You can see the options for it in `test_helper.rb` and the `braintree_blue_test` file has some examples of how to invoke it.

File: lib/active_merchant/billing/gateways/worldpay.rb (hunk 1 above)
Comment (DustinHaefele) [reply lib/active_merchant/billing/gateways/worldpay.rb:687]: No, Worldpay's docs are pretty specific about how to send the standing instruction reason on the first transaction, and this difference between their docs and our implementation is what spawned this ticket in the first place.

File: test/remote/gateways/remote_worldpay_test.rb (hunk 3 above)
Comment (DustinHaefele) [reply test/remote/gateways/remote_worldpay_test.rb]: I updated all the stored credential params in this test file to use the helper 🎉 , thanks for pointing it out!

**Pull request merged**